### PR TITLE
refactor: update core/sql.py selects to use SQL AST nodes and nodes updates.

### PR DESCRIPTION
**Pull request by Genesis929**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/core/compile/googlesql/query.py [hunk 1]
```diff
@@ -104,45 +112,44 @@ def sql(self) -> str:
...
+                f"Unsupported expression type {type(self.expression).__name__};"
+                "expected one of TableExpression, QueryExpr, str, or CTEExpression."
+            )
 
         if self.as_alias is None:
             return text
         else:
             return f"{text} {self.as_alias.sql()}"
 
+    @classmethod
```
Comment (chelsea-lin) [inline bigframes/core/compile/googlesql/query.py:138]: nit: put the `from_table_ref` before the `sql` method?

File: bigframes/core/compile/googlesql/query.py [hunk 2]
```diff
@@ -51,10 +55,14 @@ class Select(abc.SQLSyntax):
 
     select_list: typing.Sequence[typing.Union[SelectExpression, SelectAll]]
     from_clause_list: typing.Sequence[FromClause] = ()
+    distinct: bool = False
 
     def sql(self) -> str:
         text = ["SELECT"]
 
+        if self.distinct:
```
Comment (chelsea-lin) [inline bigframes/core/compile/googlesql/query.py:63]: Could you please write unit tests for the new functions in this file? I'm aiming for 100% unit-test coverage for AST compilation.

File: bigframes/core/compile/googlesql/query.py (hunk 1 above)
Comment (Genesis929) [reply bigframes/core/compile/googlesql/query.py]: Updated.

File: bigframes/core/compile/googlesql/query.py (hunk 2 above)
Comment (Genesis929) [reply bigframes/core/compile/googlesql/query.py:63]: Updated.

**Review by chelsea-lin — APPROVED** [review]
LGTM! Thanks!

**Review by TrevorBergeron — CHANGES_REQUESTED** [review]
Are we preserving identifier escaping (for column names and table expression names)?

**Comment by chelsea-lin** [comment]
> Are we preserving identifier escaping (for column names and table expression names)?
I expect more refactor PRs coming for `core/sql.py`. But you're right, the table reference should escape the identifier according to the [table naming rules](https://cloud.google.com/bigquery/docs/tables#table_naming), while the column names should be find according to the [column naming rules](https://cloud.google.com/bigquery/docs/schemas#column_names). But we can include all of them for simpler

File: bigframes/core/compile/googlesql/expression.py [hunk 3]
```diff
@@ -72,10 +73,10 @@ def __post_init__(self):
...
         text = []
         if self.project_id is not None:
-            text.append(f"`{self.project_id}`")
+            text.append(f"`{utils._escape_special_characters(self.project_id)}`")
         if self.dataset_id is not None:
-            text.append(f"`{self.dataset_id}`")
-        text.append(f"`{self.table_id}`")
+            text.append(f"`{utils._escape_special_characters(self.dataset_id)}`")
+        text.append(f"`{utils._escape_special_characters(self.table_id)}`")
         return ".".join(text)
```
Comment (chelsea-lin) [inline bigframes/core/compile/googlesql/expression.py:80]: nit: `utils._escape_special_characters(".".join(text))` only?

**Review by TrevorBergeron — APPROVED** [review]

**Review by chelsea-lin — APPROVED** [review]

File: bigframes/core/compile/googlesql/expression.py (hunk 3 above)
Comment (Genesis929) [reply bigframes/core/compile/googlesql/expression.py:79]: This may not work.

**Pull request merged**